# Plastid_Nuclear_Coevolution_Papilionoid_Legumes
## Overview
This repository contains descriptions of scripts and bioinformatic pipelines used for the manuscript **"Plastid-Nuclear Coevolution of Ribosomal Protein Genes in Papilionoid Legumes"** by Tressel et al. The study investigates plastid-nuclear coevolution by analyzing ribosomal protein genes encoded by both plastid and nuclear genomes across 52 taxa of the legume family Fabaceae. This research provides insight into how maximum likelihood methods and evolutionary rate covariation between nuclear and plastid genomes may drive cytonuclear coevolution, with implications for plant diversification and crop improvement. 

## Repository Contents
- ```Dataset_Assembly/```: 
- ```Substitution_Rate_Estimation/```:
- ```Selection_Tests/```:
- ```Evolutionary_Rate_Covariation_Analyses/```:
- ```Maximum_Likelihood_Coevolution_Analyses/```:

## Key Findings
- Elevated nonsynonymous substitution rates (*dN*) and nonsynonymous/synonymous substitution rate ratios (*dN/dS*) in plastid-encoded and nuclear-encoded plastid-targeted ribosomal protein genes.
- Significant differences in *dN* for CpRP and NuCpRP substitution rates between the 50-kb inversion clade and other legumes.
- Strong signals of cytonuclear coevolution were identified using maximum likelihood methods and evolutionary rate covariation analysis, with evidence of coordinated evolution between plastid- and nuclear-encoded ribosomal protein genes in the 50-kb clade.

## Methods
- Constraint tree contruction using RAxML
- Substitution rate estimation using PAML and LRTs
- Positive selection tests using LRTs and PAML
- Relaxed selection tests using Hyphy
- Evolutionary rate covariation analyses using normalized branch length tests to assess coevolution
- Maximum likelihood analyses to assess coevolution using PAML

## Citation
- Tressel L.G., Shrestha B., Lee C., Choi I.-S., Ruhlman T.A., Cardoso D., Wojciechowski M.F., Jansen R.K. 2024. Plastid-nuclear coevolution of ribosomal genes in papilionoid legumes.  Molecular Phylogenetics and Evolution, in review.

## Acknowledgements
We thank the TEX-LL, HUEFS, and RB herbaria for voucher deposition, the Desert Legume Program at the University of Arizona for seeds.  We also thank George Yatskievych (TEX/LL) for arranging a formal Material Transfer Agreement (Decree number 8772) under the SisGen Cadastro RDC6BE9, which facilitated research activities between our institutions. Finally, we thank Luciano P. de Queiroz (HUEFS) and Haroldo C. de Lima (RB) for providing access to living collections and for arranging field work in Brazil.

## Funding sources list
This work was supported by grants from the National Science Foundation (DEB-1853010 and DEB-1853024) to MW, RJ, and TR, the Texas Ecological Laboratory Program (EcoLab) to RJ, TR, and I-SC, the Sidney F. and Doris Blake Professorship in Systematic Botany to RJ, the CNPq (Research Productivity Fellowship no. 308244/2018-4; Universal no. 422325/2018-0), FAPESB (Universal no. APP0037/2016), and UFBA PROQUAD program to DC.
